spyder/plugins/console/widgets/main_widget.py

```diff
@@ -0,0 +1,535 @@
+            triggered=self.sig_quit_requested,
+            context=Qt.ApplicationShortcut,
+        )
+        run_action = self.create_action(
+            ConsoleWidgetActions.Run,
+            text=_("&Run..."),
+            tip=_("Run a Python script"),
+            icon=self.create_icon('run_small'),
+            triggered=self.run_script,
+        )
+        environ_action = self.create_action(
+            ConsoleWidgetActions.Environment,
+            text=_("Environment variables..."),
+            tip=_("Show and edit environment variables (for current "
+                  "session)"),
+            icon=self.create_icon('environ'),
+            triggered=self.show_env,
+        )
+        syspath_action = self.create_action(
+            ConsoleWidgetActions.SysPath,
+            text=_("Show sys.path contents..."),
+            tip=_("Show (read-only) sys.path"),
+            icon=self.create_icon('syspath'),
+            triggered=self.show_syspath,
+        )
+        buffer_action = self.create_action(
+            ConsoleWidgetActions.MaxLineCount,
+            text=_("Buffer..."),
+            tip=_("Set maximum line count"),
+            triggered=self.change_max_line_count,
+        )
+        exteditor_action = self.create_action(
+            ConsoleWidgetActions.ExternalEditor,
+            text=_("External editor path..."),
+            tip=_("Set external editor executable path"),
+            triggered=self.change_exteditor,
+        )
+        wrap_action = self.create_action(
+            ConsoleWidgetActions.ToggleWrap,
+            text=_("Wrap lines"),
+            toggled=lambda val: self.set_option('wrap', val),
+            initial=self.get_option('wrap'),
+        )
+        codecompletion_action = self.create_action(
+            ConsoleWidgetActions.ToggleCodeCompletion,
+            text=_("Automatic code completion"),
+            toggled=lambda val: self.set_option('codecompletion/auto', val),
+            initial=self.get_option('codecompletion/auto'),
+        )
+
+        # Submenu
+        internal_settings_menu = self.create_menu(
+            ConsoleWidgetMenus.InternalSettings,
+            _('Internal console settings'),
+        )
+        self.add_item_to_menu(
+            buffer_action,
+            menu=internal_settings_menu,
+            section=ConsoleWidgetInternalSettingsSubMenuSections.Main,
+        )
+        self.add_item_to_menu(
+            wrap_action,
+            menu=internal_settings_menu,
+            section=ConsoleWidgetInternalSettingsSubMenuSections.Main,
+        )
+        self.add_item_to_menu(
+            codecompletion_action,
+            menu=internal_settings_menu,
+            section=ConsoleWidgetInternalSettingsSubMenuSections.Main,
+        )
+        self.add_item_to_menu(
+            exteditor_action,
+            menu=internal_settings_menu,
+            section=ConsoleWidgetInternalSettingsSubMenuSections.Main,
+        )

```

Comment:
Please use a `for` cycle to add these actions to `internal_settings_menu`.

Reply:
Fixed